lib/Dialect/TTIR/Transforms/Passes.cpp

```diff
@@ -1453,4 +1455,121 @@ class TTIRLoadSystemDesc
   }
 };
 
+class TTIRGatherPatternMatch
+    : public ttir::impl::TTIRGatherPatternMatchBase<TTIRGatherPatternMatch> {
+public:
+  using ttir::impl::TTIRGatherPatternMatchBase<
+      TTIRGatherPatternMatch>::TTIRGatherPatternMatchBase;
+  void runOnOperation() final {
+    ModuleOp module = getOperation();
+    IRRewriter rewriter(&getContext());
+    Operation *gatherOp = nullptr;
+    bool replaceWithEmbedding = true;
+    bool reduceWeightTensorDim = false;
+
+    module->walk([&](GatherOp op) {
```

Comment:
Is there a reason we can't use pattern rewriter?

Reply:
I talked with @LPanosTT and he showed me how he wrote `UncollapsedSlidingWindow2dPatternRewriter` I'm following that and opting the pattern rewriter route.